lib/jsonapi/authorization/default_pundit_authorizer.rb

```diff
@@ -245,6 +292,23 @@ def authorize_relationship_operation(source_record, relationship_method, *args)
           ::Pundit.authorize(user, source_record, 'update?')
         end
       end
+
+      def relationship_method(data, **options)
+        relationship = data[:relationship]
+        assoc_name   = data[:relation_name]
+        prefix       = options[:prefix]
+
+        case relationship
+        when ->(relationship) { relationship.polymorphic }
+          polymorphic_type = data[:records].class.name.downcase
+          "#{prefix}_#{relationship.class_name.downcase}_#{polymorphic_type}?"
+        when ->(relationship) { relationship.class == JSONAPI::Relationship::ToOne }
```

Comment:
Yes, it seems to be so that you have to use lambda to check for the `relationshop.polymorphic` case. Ruby doesn't seize to surprise me :smile:

For the class case, though, you shouldn't need it to be a lambda.

![screen shot 2017-03-13 at 10 00 41](https://cloud.githubusercontent.com/assets/482561/23846092/5ba07308-07d4-11e7-9312-274c9a67d6e1.png)